[JOJI-769] 반응형 네비게이션

MuteAim opened the pull request:
-(현재 애니메이션 미작동 이슈 수정중입니다)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **새로운 기능**
  - 네비게이션 바가 모바일 환경에 최적화되어, 햄버거 메뉴와 확장 가능한 드롭다운이 추가되었습니다.
  - 히어로 섹션의 레이아웃 및 타이포그래피가 개선되어 다양한 화면 크기에서 최적의 사용자 경험을 제공합니다.
  - 서비스 섹션이 동적으로 서비스 항목을 렌더링하도록 변경되었습니다.
  - 포스트의 날짜 표시가 포맷팅되어 더 깔끔하게 개선되었습니다.
  - 새로운 날짜 포맷팅 기능이 추가되었습니다.
  - 앱의 스타일링과 반응형 디자인을 강화하기 위한 신규 라이브러리가 도입되었습니다.
  - 데스크탑 및 모바일 네비게이션 바 컴포넌트가 새롭게 추가되었습니다.

- **Chores**
  - 개발 환경에서 보안 강화를 위해 HTTPS 실행 설정이 업데이트되었습니다.

- **버그 수정**
  - 연락처 섹션이 제거되어 페이지 레이아웃이 조정되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MuteAim committed:
feat: 소개세션 반응형 디자인

MuteAim committed:
feat: 모바일 네비게이션 구현

MuteAim committed:
feat: 모바일 네비게이션바 개선

MuteAim committed:
feat: 중첩되는 코드 수정

MuteAim committed:
feat: 코드 재정렬

- 아직 모바일 애니메이션 작동안함

AlangGY commented on src/app/_components/navBar.js:
```diff
@@ -2,6 +2,9 @@
 import Link from "next/link";
 import Image from "next/image";
 import { useState, useEffect } from "react";
+import { FaBars, FaTimes } from "react-icons/fa";
+import { useMediaQuery } from "react-responsive";
+import { ExpandMore, ExpandLess } from "@mui/icons-material";
```
이전에 말씀드린대로 아이콘 에셋은 기존에 이미 설치되어 있는 lucide-react 라이브러리를 써주세요.

이게 은근히 에셋별로 아이콘의 스타일이 달라서, 통일감을 많이 해치게 됩니다.

AlangGY replied:
https://lucide.dev/icons/

AlangGY replied:
이번에 새롭게 추가된 패키지는 모두 제거해주세요~

AlangGY commented on src/app/_components/navBar.js:
```diff
@@ -85,135 +92,294 @@ export default function NavBar() {
-            >
-              <button
-                className={`text-sm font-semibold text-muted-foreground hover:text-primary transition-colors`}
+    <>
```
navBar jsx 구문이 지나치게 많이 복잡해진것 같습니다. mobile용과 desktop용로 컴포넌트를 분리하는 형태로 개선해주세용

AlangGY reviewed (changes requested):
개선되면 좋겠는 항목 남겨두었습니다!! 작업하시느라 고생하셨어요.

MuteAim committed:
[JOJI-775] 서비스섹션 (#35)

* feat: 서비스섹션 구현(기본구조)

* feat: 폰트크기 변경

* feat: 서비스섹션 ui개선

-생동감이 느껴지도록 하였음

* feat: 서비스섹션 ui 수정

* feat: 공지사항 페이지 ui 수정

-표시되는 시간을 YYYY-MM-DD형식으로 수정

* feat: 모바일 네비게이션 바 수정

-애니메이션이 정상 작동하도록 수정

* feat: 네비게이션 바 경로수정

* feat: contact 삭제

* feat: contact 삭제

* feat: 메뉴아이템 추출

-serviceItem.js 로 컴포넌트분리

* feat: 시간형식 변환 함수 모듈화

-useFormatDate

* feat: 포맷데이트 유틸화

- _utils/formatDate.js

MuteAim committed:
feat: 네비바 아이콘 통일

MuteAim committed:
feat: nav바 모바일/데스크톱 컴포넌트 분리

-mobileNavBar
-desktopNavBar

MuteAim commented on src/app/_components/desktopNavBar.js:
```diff
@@ -0,0 +1,220 @@
+
+const DesktopNavbar = () => {
+  const [isMenuDropdownOpen, setIsMenuDropdownOpen] = useState(false);
+  const [isUserDropdownOpen, setIsUserDropdownOpen] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false); // 로그인 여부 상태
+  const [isLoading, setIsLoading] = useState(true); //로그인 시도중인 상태
+  const [userName, setUserName] = useState(""); // 사용자 이름
+
+  useEffect(() => {
+    fetch(`${process.env.NEXT_PUBLIC_BASE_URL}/auth/profile`, {
+      method: "GET",
+      credentials: "include",
+    })
+      .then((response) => {
+        if (response.status === 200) {
+          return response.json();
+        } else if (response.status === 401) {
+          throw new Error("Unauthorized");
+        } else {
+          throw new Error("Unexpected error");
+        }
+      })
+      .then((data) => {
+        setIsLoggedIn(true);
+        setUserName(data.name);
+      })
+      .catch((error) => {
+        if (error.message === "Unauthorized") {
+          setIsLoggedIn(false);
+        }
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
```
그래 안그래도 그럴려고

MuteAim committed:
feat: 로그인 기능 모듈화

-useAuth 커스텀훅으로 구현

AlangGY commented on src/app/_components/navBar.js:
```diff
@@ -2,9 +2,8 @@
 import { useState, useEffect } from "react";
-import { FaBars, FaTimes } from "react-icons/fa";
 import { useMediaQuery } from "react-responsive";
-import { ExpandMore, ExpandLess } from "@mui/icons-material";
```
package.json에서 react-icons/fa, @mui/icons-material 패키지를 제거하고, 다시 npm i 해서 커밋해주세용

AlangGY commented on src/app/_components/desktopNavBar.js:
```diff
@@ -9,57 +10,10 @@ const MENU_BUTTON_HEIGHT = 20;
-        setIsLoading(false);
-      });
-  }, []);
+  const { isLoggedIn, isLoading, userName, redirectToLogin, handleLogout } =
+    useAuth();
```
useAuth hook의 상태들은 지역스코프로 관리되는 상태이기 때문에 DesktopNavbar <-> MobileNavbar로 전환될때마다 매번 호출이 일어날 가능성이 있어요. 이 둘이 같은 상태를 공유하도록 하는 작업이 필요합니다. 요건 따로 설명드릴게요

AlangGY reviewed (changes requested):
크 모듈화의 황제가 되어가고 있는 모습이 보이는 군요.
단 한가지 문제점이 있어서, 한 단계 더 레벨업 시켜드리겠습니다

MuteAim committed:
feat: 네비바 상태공유

-props로 같은상태 공유

MuteAim committed:
feat: props 컨벤션 수정

AlangGY reviewed (approved):
아따 믓집니다잉

MuteAim committed:
Update page.js

MuteAim committed:
Merge branch 'development' into feat/JOJI-797/소개섹션-반응형
